### PR TITLE
fix: static validation error and BCP error in `avm.res.dev-ops-infrastructure.pool`

### PR DESCRIPTION
**Pull request by johnlokerse**
## Description

This pull request fixes the following:
* The static validation is now passing, fixed by rerunning the Set-AVMModule command
* BCP error regarding a property/parameter that cannot be null - fixed this by making it required

Tagging @matebarabas @AlexanderSehr 

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.res.dev-ops-infrastructure.pool](https://github.com/johnlokerse/bicep-registry-modules/actions/workflows/avm.res.dev-ops-infrastructure.pool.yml)     |

## Type of Change

- [x] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/dev-ops-infrastructure/pool/main.bicep [hunk 1]
```diff
@@ -293,7 +293,7 @@ type organizationProfileType = {
   @description('Optional. The type of permission which determines which accounts are admins on the Azure DevOps pool.')
   permissionProfile: {
     @description('Optional. Determines who has admin permissions to the Azure DevOps pool.')
```
Comment (ReneHezser) [inline avm/res/dev-ops-infrastructure/pool/main.bicep:295]: If I am not mistaken, the value is now mandatory. Therefore, Optional should be Required.

Comment (johnlokerse) [reply avm/res/dev-ops-infrastructure/pool/main.bicep:295]: Yes, correct. Thanks for checking @ReneHezser!

Comment (johnlokerse) [reply avm/res/dev-ops-infrastructure/pool/main.bicep:295]: Fixed

**Review by eriqua — APPROVED** [review]
🆗️

**Comment by matebarabas** [comment]
@eriqua, @johnlokerse, the test deployments seem to have failed (the module is not published): https://github.com/Azure/bicep-registry-modules/actions/runs/10950416993

**Comment by johnlokerse** [comment]
@matebarabas @eriqua This is error comes up due to incorrect quotas in the region that the MDP resource is deployed to. I know Alexander has tested a deployment on his side. I believe he tested it in northeurope, due to capacity limits in westeurope.  

See docs: https://learn.microsoft.com/en-us/azure/devops/managed-devops-pools/prerequisites?view=azure-devops&tabs=azure-portal#create-a-quota-support-request

What is the best way to approach this? Hardcode the deployment location if we are sure which location @AlexanderSehr has deployed to?

**Comment by eriqua** [comment]
On it @johnlokerse

**Comment by eriqua** [comment]
Hey @johnlokerse, I'm retrieving a list of allowed locations for our target environment. 
For now, I tried to enforce one of those (uksouth) through the pipeline, using the customLocation feature.
While the defaults test is successful, max and waf-aligned tests are failing due to UnauthorizedAccessToVirtualNetwork. It seems that the DevOpsInfrastructure service principal does not have Read access to the virtual network, and would require Reader and Network Contributor access. 
Does that sound familiar? 
Did you have to set that up in your testing environment?

Let me know if that and anything else so that we can test a successful deployment of all e2e tests before enforcing specific locations in code as a next step.

**Comment by johnlokerse** [comment]
Hey @eriqua, the permissions on the virtual network are set through the `dependencies.bicep` as seen below:

```bicep
// Network Contributor role assignment
resource roleAssignments 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
  name: guid(subscription().subscriptionId, 'DevOpsInfrastructure', 'Network Contributor', 'max')
  properties: {
    principalId: devOpsInfrastructureObjectID // DevOpsInfrastructure service principal
    #disable-next-line use-resource-id-functions
    roleDefinitionId: '/providers/Microsoft.Authorization/roleDefinitions/4d97b98b-1d4f-4787-a291-c67834d212e7'
    principalType: 'ServicePrincipal'
  }
  scope: virtualNetwork
}
```

The deployment uses the new CI secret feature. My validation deployments are done through that way. As far as I can see it is also the case in the AVM test deployment (https://github.com/Azure/bicep-registry-modules/actions/runs/10963081104/job/30458820341#step:4:542): 

CleanShot 2024-09-23 at 10 26 31@2x

@eriqua What are the deployment permission differences between Default and Max or WAF? The AVM validation deployment has a successful Default deployment, which also deploys an agent pool in Azure DevOps.

Below the output of a Defaults deployment (agent pool in AzDo):
CleanShot 2024-09-23 at 10 49 00@2x

**Comment by eriqua** [comment]
Hi @johnlokerse thanks for the above details and sorry if I'm not aligned if discussions already happened prior to raising this PR.

As you know, as per the new CI secrets feature, we should also set up the brm environment in the same way you did for yours. That includes setting up secrets in the BRM keyvault. I see we have secrets created in our keyvault too, but we are missing values. 
I'll set up the DevOpsInfrastructure one, but which values should we set up for the other 2 secrets ado organization and project? 
Also, do you know if ado permissions have been set up already for our deployment spn?

I'd appreciate if we could connect offline so that I can support setting the above in order to publish this module.

**Comment by eriqua** [comment]
🚀 The module is published 🚀 

Just providing updates here for future reference: 
- `InsufficientCoreQuota` error: Solved via the `enforcedLocation` (uksouth) workaround in code. Ref PR #3351 
- `UnauthorizedAccessToVirtualNetwork` error: Custom CI secrets have been updated in upstream key vault. In particular, the CI-DevOpsInfrastructureObjectID value.
- `PoolProvisioningFailed` (SPN missing create permissions) error: In ADO, the SPN has been added to the `Project collection administrator` group. Setting the SPN Administrator on agent pools at Organizational settings level (least privilege) as per documentation was not enough for max and waf-aligned tests.

**Comment by AlexanderSehr** [comment]
> * create permissions) error: In ADO, the SPN has been added to the `Project collection administrator` group. Setting the SPN Administrator on agent pools at Organizational settings level (least privilege) as per documentation was not enough for max and waf-aligned tests.

Hey @eriqua & @johnlokerse, was #3 added to the test/documentation by any chance?

**Comment by eriqua** [comment]
> > * create permissions) error: In ADO, the SPN has been added to the `Project collection administrator` group. Setting the SPN Administrator on agent pools at Organizational settings level (least privilege) as per documentation was not enough for max and waf-aligned tests.
> 
> Hey @eriqua & @johnlokerse, was #3 added to the test/documentation by any chance?

Hey @AlexanderSehr I believe there are conversations ongoing with the PG to investigate why documented permissions were not enough. @johnlokerse please chime in if any news.
Depending on the outcome of that discussion, I agree it would be beneficial to document it

**Comment by johnlokerse** [comment]
@AlexanderSehr @eriqua I have not heard anything back yet. I will post something here when I hear something (or just add it to the docs ;-))

**Comment by AlexanderSehr** [comment]
I see. Thanks for elaborating 💪

**Pull request merged**